go/api/handlers/document.go

```diff
@@ -0,0 +1,189 @@
+package handlers
+
+import (
+	"challenge-flutter-go/api/errorHandlers"
+	"challenge-flutter-go/api/responses"
+	"challenge-flutter-go/api/utils"
+	"challenge-flutter-go/models"
+	"challenge-flutter-go/repository"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+)
+
+type DocumentHandler struct {
+	Repository     repository.DocumentRepository
+	TripRepository repository.TripRepository
+}
+
+// @Summary		Get all documents of a trip
+// @Description	Get all documents of a trip
+// @Tags			document
+// @Accept			json
+// @Produce		json
+// @Security		BearerAuth
+// @Param			id		path		string	true	"ID of the trip"
+// @Failure		400		{object}	error
+// @Failure		401		{object}	error
+// @Router			/trips/{id}/documents [get]
+func (handler *DocumentHandler) GetDocumentsOfTrip(context *gin.Context) {
+	tripId := context.Param("id")
+
+	trip, _ := handler.TripRepository.Get(tripId)
+
+	documents, err := handler.Repository.GetDocuments(trip)
```

Comment:
```suggestion
	documents, err := handler.Repository.GetDocuments(tripId)
```

Pas besoin d'aller recherchee le trip pour le passer au repo, on est déjà assuré qu'il exsite dans les middlewares. Il faut modifier le repo aussi.